test/Microsoft.DotNet.Configurer.UnitTests/GivenADotnetFirstTimeUseConfigurer.cs

```diff
@@ -258,6 +271,26 @@ public void It_prints_the_first_time_use_notice_if_the_cache_sentinel_exists_but
             _reporterMock.Verify(r => r.Write(It.IsAny<string>()), Times.Never);
         }
 
+        [Fact]
+        public void It_adds_executable_package_path_to_environment_path_when_the_first_notice_sentinel_does_not_exist()
```

Comment:
I am worried about the case of the first notice sentinel never being created and us executing this code all the time, without really needing to do so.

Reply:
If the PATH is added already, it will be no op, and checking env is relativity fast (we are checking at least 5 env var at start up time already). And I cannot think up a case that on Windows, adding to registry will fail constantly. Since we are adding a user path to a user env,

Reply:
Well, we are treating performance right now with the utmost importance and it feels weird to me that we are adding something to the critical code path that will run for every single donet command, even if we don't need it. Wouldn't it make sense to skip it once we know we can do just that?

Reply:
Same as the comment above, we have similar call multiple times existing today. If it doesn't show up in our performance radar, it is probably not the bottle neck. However, if we want really want avoid this check, I can put it into different place(for timing) with checking OS.